File: Basic Algorithm Scripting/Python/findLongestWordLength.py
# Return the length of the longest word in the provided sentence.
# Your response should be a number.


def findLongestWord(string: str):
    return max([len(word) for word in string.split(" ")])


# Extra Credit: Returns the longest word.
def findLongestWord(string: str):
    return [
        word
        for word in string.split(" ")
        if len(word) == max([len(word) for word in string.split(" ")])
    ]

File: Basic Algorithm Scripting/Javascript/findLongestWordLength.js
/*
Return the length of the longest word in the provided sentence.
Your response should be a number.
*/

function findLongestWordLength(str) {
    let wordArr = str.trim().split(/\s/);
    let lenArr = [];
    for (let x = 0; x < wordArr.length; x++) {
      lenArr.push(wordArr[x].length);
    }
    return Math.max(...lenArr);
  }